test/integration/include/findAll.test.js

```diff
@@ -1961,5 +1961,92 @@ describe(Support.getTestDialectTeaser('Include'), function() {
         expect(parseInt(post["comments.commentCount"], 10)).to.equal(3);
       });
     });
+
+    it('Should return posts with nested include with inner join with a m:n association', function () {
+      var User = this.sequelize.define('User', {
+        username: {
+          type: DataTypes.STRING,
+          primaryKey: true
+        }
+      })
+        , Entity = this.sequelize.define('Entity', {
+          entity_id: {
+            type: DataTypes.INTEGER,
+            autoIncrement: true,
+            primaryKey: true
+          },
+          creator: {
+            type: DataTypes.STRING,
+            allowNull: false
+          },
+          votes: {
+            type: DataTypes.INTEGER,
+            allowNull: false,
+            defaultValue: 0
+          }
+        })
+        , Post = this.sequelize.define('Post', {
+          post_id: {
+            type: DataTypes.INTEGER,
+            allowNull: false,
+            primaryKey: true
+          }
+        })
+        , TaggableSentient = this.sequelize.define('TaggableSentient', {
+          nametag: {
+            type: DataTypes.STRING,
+            primaryKey: true
+          }
+        });
+
+      Entity.belongsTo(User, { foreignKey: 'creator', targetKey: 'username' });
+      Post.belongsTo(Entity, { foreignKey: 'post_id', targetKey: 'entity_id' });
+
+      Entity.belongsToMany(TaggableSentient, { 
+        as: 'tags', 
+        through: { model: 'EntityTag', unique: false }, 
+        foreignKey: 'entity_id', 
+        otherKey: 'tag_name' 
+      });
+
+      TaggableSentient.belongsToMany(Entity, { 
+        as: 'tags', 
+        through: { model: 'EntityTag', unique: false }, 
+        foreignKey: 'tag_name', 
+        otherKey: 'entity_id' 
+      });
+
+      return this.sequelize.sync({ force: true }).then(function () {
+        return Post.findAll({
+            include: [{
+              model: Entity,
+              required: true,
+              include: [{
+                model: User,
+                required: true
+              },{
+                model: TaggableSentient,
+                as: 'tags',
+                required: true,
+                through: {
+                  where: {
+                    tag_name: ['tag1', 'tag2']
+                  }
+                }
+              }]
+            }],
+            limit: 5,
+            offset: 0,
+            order: [
+              [ { raw: '\`Entity.votes\`' }, 'DESC' ]
```

Comment:
@mickhansen I have to use raw / sequelize.literal as there is problem with when using
```js
order: [
    [ this.sequelize.col('Entity.votes'), 'DESC' ]
]
```
I'll raise a separate issue for the same.